### PR TITLE
Add tests & fixes for AlgorandExplorer

### PR DESCRIPTION
**Pull request by haammar-ledger**

### CODE REVIEW

File: ledger-core-algorand/test/AlgorandTestFixtures.hpp [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    const model::AssetParams testAsset() {
+        static model::AssetParams asset;
+
+        if (asset.assetName.isEmpty()) {
+            asset.assetName = "Obelix Coin";
+            asset.unitName = "OBC";
+            asset.total = 1000000;
+            asset.decimals = 3;
+            asset.defaultFrozen = false;
+            asset.creatorAddr = Address("RGX5XA7DWZOZ5SLG4WQSNIFKIG4CNX4VOH23YCEX56523DQEAL3QL56XZM");
```
Comment (barjonr) [inline ledger-core-algorand/test/AlgorandTestFixtures.hpp:71]: you could use `OBELIX_ADDRESS`

File: ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp [hunk 2]
```diff
@@ -300,59 +306,86 @@ namespace constants {
...
+            if (node.HasMember(constants::xParticipation.c_str())) {
+                account.participation = model::KeyRegTxnFields();
+                parseParticipationInfo(node[constants::xParticipation.c_str()].GetObject(), *account.participation);
             }
         }
 
     private:
 
+        template <class T>
+        static void getMandatoryStringField(const T & node, const std::string & fieldName, Option<std::string> & field) {
```
Comment (barjonr) [inline ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp:346]: Wouldn't it be possible to use `getMandatoryStringField(const T&, const std::string&, std::string&)` ?
Something like
`template <class T>
static void getMandatoryStringField(const T & node, const std::string & fieldName, Option<std::string> & field) { getMandatoryStringField(node, fieldName, *field); }`
This way there is no duplicated code.

File: ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp [hunk 3]
```diff
@@ -130,37 +141,31 @@ namespace constants {
...
+            getMandatoryUint64Field(node, constants::xAmount, assetAmount.amount);
+            getMandatoryBoolField(node, constants::xFrozen, assetAmount.frozen);
         }
 
         template <class T>
         static void parseAssetsAmounts(const T& node, std::map<uint64_t, model::AssetAmount> & assetsAmounts) {
             for (rapidjson::Value::ConstMemberIterator child = node.MemberBegin(); child != node.MemberEnd(); ++child) {
                 model::AssetAmount assetAmount;
-                const auto coinId = child->name.GetUint64();
+                const auto coinId = BigInt::from_dec(child->name.GetString())->toUint64();
```
[the remark this answers is not in the document]
Comment (haammar-ledger) [reply ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp:168]: Maybe... dumb question, but are we sure that `uint64_t` and `unsigned long long` will always get along?
I personally have no preference, I'm just assuming `BigInt` handles the type for us but I'm not even sure.

[the remark this answers is not in the document]
Comment (haammar-ledger) [reply ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp:168]: This SO answer suggests that `uint64_t` and `unsigned long long` are not the same :/ https://stackoverflow.com/a/32198541/1160527

[the remark this answers is not in the document]
Comment (barjonr) [reply ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp:168]: The standard requires an unsigned long long to be at least 64 bits long, so a uint64_t will always fit in an unsigned long long. (https://en.cppreference.com/w/cpp/language/types)

File: ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp (hunk 2 above)
Comment (haammar-ledger) [reply ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp:346]: The `Option` field has to be initialized before being passed as `*field`.
If I do it that way it looks like this:
```
        template <class T>
        static void getMandatoryStringField(const T & node, const std::string & fieldName, Option<std::string> & field) {
            assert(node.HasMember(fieldName.c_str()));
-           field = node[fieldName.c_str()].GetString();
+           field = Option<string>();
+           field = getMandatoryStringField(node, fieldName, *field);
        }

```
It's a bit more verbose. Also the test (`if` / `assert`) is done twice. Do you like it better?

Comment (barjonr) [reply ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp:346]: I thought the Option would already be initialized... Don't bother then I guess

File: ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (haammar-ledger) [reply ledger-core-algorand/inc/algorand/AlgorandJsonParser.hpp:168]: Nice! Better reference than SO :D
I'll do the change then. Thanks.

**Pull request merged**